From: Larry Finger <Larry.Finger@lwfinger.net>

In the SSB SPROM a field set to all ones means the value
is not defined in the SPROM.
In case of the boardflags, we need to set them to zero
to avoid confusing drivers. Drivers will only check the
flags by ANDing.

This patch is in wireless-testing.git, commit
d30cdf8a251e88e58bc566f5acaa9eb97ac102e9

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
Signed-off-by: Gabor Stefanik <netrolller.3d@gmail.com>
Signed-off-by: Michael Buesch <mb@bu3sch.de>


Index: linux-2.6.25/drivers/ssb/pci.c
===================================================================
--- linux-2.6.25.orig/drivers/ssb/pci.c	2008-04-17 17:11:28.000000000 +0200
+++ linux-2.6.25/drivers/ssb/pci.c	2008-04-19 17:56:54.000000000 +0200
@@ -482,6 +482,11 @@ static int sprom_extract(struct ssb_bus 
 			goto unsupported;
 	}
 
+	if (out->boardflags_lo == 0xFFFF)
+		out->boardflags_lo = 0;  /* per specs */
+	if (out->boardflags_hi == 0xFFFF)
+		out->boardflags_hi = 0;  /* per specs */
+
 	return 0;
 unsupported:
 	ssb_printk(KERN_WARNING PFX "Unsupported SPROM revision %d "
